app/upgrades/v20/upgrades.go

```diff
@@ -25,7 +46,80 @@ func CreateUpgradeHandler(
 			return vm, err
 		}
 
+		InitializeMaxProviderConsensusParam(ctx, keepers.ProviderKeeper)
+		InitializeMissingValidatorSetCaps(ctx, keepers.ProviderKeeper)
+		SetMaxValidators(ctx, *keepers.StakingKeeper)
+		InitializeLastProviderConsensusValidatorSet(ctx, keepers.ProviderKeeper, *keepers.StakingKeeper)
+
 		ctx.Logger().Info("Upgrade v20 complete")
 		return vm, nil
 	}
 }
+
+// InitializeMissingValidatorSetCaps initializes the max validators
+// parameter for existing consumers to the MaxProviderConsensusValidators parameter.
+// This is necessary to avoid those consumer chains having an excessive amount of validators.
```

Comment:
Is this needed? If they don't allow inactive validators, then they cannot have more than 180 anyway, right? My concern is that if new consumers joins before the v20 upgrade, this will set their valset cap to 180 regardless of their power shaping config.

Reply:
Good point. Yeah, I wanted this as another "line of defense", but it's not strictly necessary anymore, since inactive validators will by default not be allowed.
Seems better to remove this